### PR TITLE
[BTRX-689] Enforce backend roles

### PR DESCRIPTION
**Pull request by tadeork**
## Addresses
https://broadinstitute.atlassian.net/browse/BTRX-689

## Changes
Filters access to functionalities when the user is Read Only Admin.

## Testing
Describe for the reviewer how to test the specifics of this PR, both positive and negative cases.

---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from Belatrix
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: grails-app/controllers/org/broadinstitute/orsp/ReadOnlyInterceptor.groovy [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.orsp
+
+import grails.converters.JSON
+
+/**
+ * Interceptor specific to API access related to the Read Only admin role
+ *
+ * Requires not to be ro_admin role to proceed
+ */
+class ReadOnlyInterceptor implements UserInfo {
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/ReadOnlyInterceptor.groovy]: Can this work with [`@CompileStatic`](https://objectcomputing.com/resources/publications/sett/september-2015-grails-3-interceptors) ?

Comment (tadeork) [reply grails-app/controllers/org/broadinstitute/orsp/ReadOnlyInterceptor.groovy]: No, I just tried and didn't work.

Comment (rushtong) [reply grails-app/controllers/org/broadinstitute/orsp/ReadOnlyInterceptor.groovy]: 👍

Comment (rushtong) [reply grails-app/controllers/org/broadinstitute/orsp/ReadOnlyInterceptor.groovy]: @tadeork - I made the `@CompileStatic` change to Vero's branch that I'm testing and it seems to work fine.

**Pull request merged**